### PR TITLE
Add support for Simple reStructuredText Tables

### PR DESCRIPTION
**Pull request by simongibbons**
This is a first stab at getting an implementation for reading and writing [Simple RST tables](http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#simple-tables), as suggested in #4590 

Currently there is no real error checking in the code, and there is no support for an overflowing right hand column, these will be added in due course.

### CODE REVIEW

**Comment by simongibbons** [comment]
Ok I've added support for the overflowing right hand column now.

**Comment by simongibbons** [comment]
The appveyor failure doesn't look like it's anything to do with me, just conda failing to install python.

File: astropy/io/ascii/rst.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+class SimpleRSTHeaderSplitter(DefaultSplitter):
+    delimiter = ' '
+
+class SimpleRSTHeader(FixedWidthHeader):
+    position_line = 0
+    start_line = 1
+    splitter_class = SimpleRSTHeaderSplitter
+    position_char = '='
+
+class SimpleRSTDataSplitter(FixedWidthTwoLineDataSplitter):
```
Comment (taldcroft) [inline astropy/io/ascii/rst.py]: I think you can avoid making this custom splitter class by the following (not tested):
```
class SimpleRSTHeader(FixedWidthHeader):
    # (stuff you already have)
    def get_cols(self, lines):
        super(SimpleRSTHeader, self).get_cols(lines)

        # Right hand column can be unbounded
        self.cols[-1].end = None
```

File: astropy/io/ascii/__init__.py [hunk 2]
```diff
@@ -40,6 +40,7 @@
 from .fixedwidth import (FixedWidth, FixedWidthNoHeader,
                          FixedWidthTwoLine, FixedWidthSplitter,
                          FixedWidthHeader, FixedWidthData)
+from .rst import (SimpleRST)
```
Comment (taldcroft) [inline astropy/io/ascii/__init__.py]: No need for parentheses here.

File: astropy/io/ascii/rst.py (hunk 1 above)
Comment (simongibbons) [inline astropy/io/ascii/rst.py]: Yes you are right, It's cleaner IMO to instead modify `get_fixedwidth_params` in `SimpleRSTHeader` to deal with the unbounded right hand column. This also has the benefit of allowing the right hand column name to be unbounded also (to fit with the spec)

**Comment by taldcroft** [comment]
@simongibbons - looking pretty good now, your implementation of the unbounded right most column is fine by me.  One suggestion now is testing that right-indenting the entire table and having one or more trailing spaces in the row-definition column doesn't cause any problem.

**Comment by simongibbons** [comment]
Looks like those tests work fine, what's left to do now? Just update the docs and CHANGES?

File: astropy/io/ascii/rst.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        return vals, starts, ends
+
+class SimpleRSTData(FixedWidthData):
+    start_line = 3
+    end_line = -1
+    splitter_class = FixedWidthTwoLineDataSplitter
+
+class SimpleRST(FixedWidth):
+    """
+    Read or write a simple reStructuredText table
```
Comment (taldcroft) [inline astropy/io/ascii/rst.py]: This docstring should be more detailed, including a link to http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#simple-tables for a description of what is meant.  It should have an example table as well as a description of what is not supported (e.g. a secondary line of `------ ----` in order to indicate column spans).

**Comment by taldcroft** [comment]
Overall looking quite close to finished.  It needs an entry in the CHANGES.rst new features for 1.2.

Also, I think that the class and reader name should be tweaked a bit.  I like to think hierarchically and "Simple" is not the most important thing here.  So to me it makes sense to use `RSTSimple` or maybe `RstSimple` with reader format name in both cases being `rst_simple`.  That way later on there might be an `RstGrid`.

Another possibility that might be even better is to just call this the `RST` reader/writer, and allow for future expansion of the capabilities of that class, e.g. to support the grid format.  The documentation would clearly indicate that the implementation currently supports only the simple format.   Then we would eventually need a keyword to differentiate simple vs. grid (at least for writing), but this has the benefit of keeping the top-level list of reader classes as succinct as possible.  For now you would still have the description `Simple reStructuredText format table` in the main table to make it clear right away that the implementation is not complete for all RST table types.

**Comment by taldcroft** [comment]
Finally, for code that is mostly all new and took a few detours along the way, I would suggest to squash down some or most of the commits.  Use `git rebase -i ...`.

**Comment by simongibbons** [comment]
Ok that should be all the changes that you've asked for, if the CI checks pass then this should be ready for a final code review and merging. Thanks for all the help in getting this ready @taldcroft.

**Comment by taldcroft** [comment]
Looks good, thanks @simongibbons !

**Pull request merged**